client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -1214,4 +1227,13 @@ const WrappedPlanFeatures2023Grid = ( props: PlanFeatures2023GridType ) => {
 	);
 };
 
-export default WrappedPlanFeatures2023Grid;
+export default withSelect( ( select: SelectFunction, ownProps: PlanFeatures2023GridType ) => {
+	const locale = getLocaleSlug();
+	locale && ( select( PLANS_STORE ) as PlansSelect ).getSupportedPlans( locale );
+	const datastorePlansLoaded = ( select( PLANS_STORE ) as PlansSelect ).hasFinishedResolution(
+		'getSupportedPlans',
+		[ locale ]
+	);
+
+	return { ...ownProps, datastorePlansLoaded };
+} )( WrappedPlanFeatures2023Grid );
```

Comment:
This isn't necessary, since `PlanFeaturesMain` is connected, but no harm in keeping if to ensure this isn't imported and used anywhere in a broken state. It's also the only properly typed usage.